rest_framework/serializers.py

```diff
@@ -1081,7 +1081,10 @@ def save_object(self, obj, **kwargs):
                     self.save_object(related)
                 else:
                     # Reverse FK or reverse one-one
-                    setattr(obj, accessor_name, related)
+                    try:
+                        setattr(obj, accessor_name, related)
+                    except ValueError:
+                        getattr(obj, accessor_name).add(*related, bulk=False)
```

Comment:
I am afraid this is not so easy, as it is not because of only our code or how we use DRF.
M2M relations are possible in Rest Framework but now (https://docs.djangoproject.com/en/1.9/releases/1.9/#models) in 1.9 they added `batch=True` kwarg in `.add()` method of m2m relations. 
I am afraid it will fail for all m2m relations.